core/src/test/java/org/elasticsearch/common/xcontent/ConstructingObjectParserTests.java

```diff
@@ -174,10 +175,10 @@ public void testRepeatedConstructorParam() throws IOException {
                 + "  \"vegetable\": 2\n"
                 + "}");
         Throwable e = expectThrows(ParsingException.class, () -> randomFrom(HasCtorArguments.ALL_PARSERS).apply(parser, MATCHER));
-        assertEquals("[has_required_arguments] failed to parse field [vegetable]", e.getMessage());
+        assertEquals("[has_required_arguments] failed to parse object", e.getMessage());
         e = e.getCause();
-        assertThat(e, instanceOf(IllegalArgumentException.class));
-        assertEquals("Can't repeat param [vegetable]", e.getMessage());
+        assertThat(e, instanceOf(JsonParseException.class));
```

Comment:
So this is one problem with doing this as a read-on-startup property. We can no longer test things like this. This test is fairly worthless unless the property is `false` and we have to manually perform the duplication detection.

I think it'd be worth leaving a comment about that.